yarn-project/sequencer-client/src/block_builder/solo_block_builder.ts

```diff
@@ -280,40 +300,50 @@ export class SoloBlockBuilder implements BlockBuilder {
     return [rootOutput, rootProof];
   }
 
-  protected async validatePartialState(partialState: PartialStateReference) {
-    await Promise.all([
-      this.validateSimulatedTree(
-        await this.getTreeSnapshot(MerkleTreeId.NOTE_HASH_TREE),
-        partialState.noteHashTree,
-        'NoteHashTree',
-      ),
-      this.validateSimulatedTree(
-        await this.getTreeSnapshot(MerkleTreeId.NULLIFIER_TREE),
-        partialState.nullifierTree,
-        'NullifierTree',
-      ),
-      this.validateSimulatedTree(
-        await this.getTreeSnapshot(MerkleTreeId.CONTRACT_TREE),
-        partialState.contractTree,
-        'ContractTree',
-      ),
-      this.validateSimulatedTree(
-        await this.getTreeSnapshot(MerkleTreeId.PUBLIC_DATA_TREE),
-        partialState.publicDataTree,
-        'PublicDataTree',
-      ),
-    ]);
+  protected validatePartialState(
+    partialState: PartialStateReference,
+    treeSnapshots: Map<MerkleTreeId, AppendOnlyTreeSnapshot>,
+  ) {
+    this.validateSimulatedTree(
+      treeSnapshots.get(MerkleTreeId.NOTE_HASH_TREE)!,
+      partialState.noteHashTree,
+      'NoteHashTree',
+    );
+    this.validateSimulatedTree(
+      treeSnapshots.get(MerkleTreeId.NULLIFIER_TREE)!,
+      partialState.nullifierTree,
+      'NullifierTree',
+    );
+    this.validateSimulatedTree(
+      treeSnapshots.get(MerkleTreeId.CONTRACT_TREE)!,
+      partialState.contractTree,
+      'ContractTree',
+    );
+    this.validateSimulatedTree(
+      treeSnapshots.get(MerkleTreeId.PUBLIC_DATA_TREE)!,
+      partialState.publicDataTree,
+      'PublicDataTree',
+    );
   }
 
   protected async validateState(state: StateReference) {
-    await Promise.all([
+    const promises = [
+      MerkleTreeId.NOTE_HASH_TREE,
+      MerkleTreeId.CONTRACT_TREE,
+      MerkleTreeId.NULLIFIER_TREE,
+      MerkleTreeId.PUBLIC_DATA_TREE,
+    ].map(async (id: MerkleTreeId) => {
+      return { key: id, value: await this.getTreeSnapshot(id) };
+    });
+    const snapshots: Map<MerkleTreeId, AppendOnlyTreeSnapshot> = new Map(
+      (await Promise.all(promises)).map(obj => [obj.key, obj.value]),
+    );
+    this.validatePartialState(state.partial, snapshots),
       this.validateSimulatedTree(
         await this.getTreeSnapshot(MerkleTreeId.L1_TO_L2_MESSAGE_TREE),
         state.l1ToL2MessageTree,
         'L1ToL2MessageTree',
-      ),
-      this.validatePartialState(state.partial),
-    ]);
+      );
```

Comment:
I think this should be `;` separating these calls and not a `,`
```suggestion
    this.validatePartialState(state.partial, snapshots);
      this.validateSimulatedTree(
        await this.getTreeSnapshot(MerkleTreeId.L1_TO_L2_MESSAGE_TREE),
        state.l1ToL2MessageTree,
        'L1ToL2MessageTree',
      );
```